filebeat/module/iis/w3svc/ingest/pipeline.json

```diff
@@ -0,0 +1,46 @@
+{
+  "description": "Pipeline for parsing iis w3svc logs",
+  "processors" : [
+    {
+      "grok" : {
+        "field" : "message",
+        "patterns" : [
+          "%{TIMESTAMP_ISO8601:iis.access.time} %{IPORHOST:iis.access.site} %{WORD:iis.access.method} %{URIPATH:iis.access.url} %{NOTSPACE:iis.access.querystring} %{NUMBER:iis.access.port} %{NOTSPACE:iis.access.user_name} %{IPORHOST:iis.access.remote_ip} %{NOTSPACE:iis.access.agent} ?%{NOTSPACE:iis.access.referrer}? %{NUMBER:iis.access.response_code} %{NUMBER:iis.access.subresponse_code} %{NUMBER:iis.access.scstatus} %{NUMBER:iis.access.response_time}"
```

Comment:
This `grok` pattern looks pretty good to me. However, I would like to propose a couple of changes:

1. I believe that the second field in the IIS log files is the IP address of the server hosting IIS, not the site configured within IIS that generated the log files. I therefore think that the field `iis.access.site` should be renamed to `iis.access.server_ip`. Note that IIS can be configured to output additional fields in log files. One of these additional fields is the IIS site. Therefore, if this IIS Filebeat module is ever going to support the "full" log file output, then `iis.access.site` must be renamed to `iis.access.server_ip` to avoid confusion or field name conflicts.

2. For consistency with other fields, words should be separated by spaces. Therefore, can `iis.access.querystring` be renamed `iis.access.query_string`.

3. The documentation for contributing to Beats suggests using suffixes to indicate what units a value is in. Therefore, can `iis.access.response_time` be renamed `iis.access.time_taken_ms`? Note that I've renamed this to time taken, rather than response time, for consistency with the IIS documentation.

4. I'm not sure where the names for the fields `iis.access.subresponse_code` and `iis.access.scstatus` have come from. For better clarity on how these fields relate to fields outputted by IIS, I would suggest calling these fields `iis.access.sub_status` and `iis.access.win32_status` respectively.

Reply:
@BongoEADGC6 I agree with @cgwrench points above.